mkt/reviewers/views.py

```diff
@@ -1213,3 +1213,44 @@ class ReviewerScoreViewSet(CORSMixin, MarketplaceView, viewsets.ModelViewSet):
     # queryset to prevent instances with other note_key values from ever being
     # returned.
     queryset = ReviewerScore.objects.filter(note_key=mkt.REVIEWED_MANUAL)
+
+
+@reviewer_required
+def moderatelog(request):
+    form = ModerateLogForm(request.GET)
+    modlog = ActivityLog.objects.editor_events()
+    if form.is_valid():
+        if form.cleaned_data['start']:
+            modlog = modlog.filter(created__gte=form.cleaned_data['start'])
+        if form.cleaned_data['end']:
+            modlog = modlog.filter(created__lt=form.cleaned_data['end'])
+        if form.cleaned_data['search']:
+            modlog = modlog.filter(action=form.cleaned_data['search'].id)
+
+    pager = paginate(request, modlog, 50)
+    data = context(request, form=form, pager=pager, tab='moderatelog')
+    return render(request, 'reviewers/moderatelog.html', data)
+
+
+@reviewer_required
+def moderatelog_detail(request, id):
+    log = get_object_or_404(ActivityLog.objects.editor_events(), pk=id)
+    review = None
+
+    if len(log.arguments) > 1 and isinstance(log.arguments[1], Review):
+        review = log.arguments[1]
+    is_admin = acl.action_allowed(request, 'ReviewerAdminTools', 'View')
+    can_undelete = review and review.deleted and (
+                    is_admin or request.user.pk == log.user.pk)
+
+    if request.method == 'POST':
+        # A Form seems overkill for this.
+        if request.POST['action'] == 'undelete':
+            if not can_undelete:
+                raise PermissionDenied
+            ReviewerScore.award_moderation_points(
+                log.user, review.addon, review.id, undo=True)
+            review.undelete()
+            return redirect('reviewers.apps.moderatelog.detail', id)
+    data = context(request, log=log, review=review, can_undelete=can_undelete)
```

Comment:
It feels like there's about 5 code paths in here and I only see one test. Another reason forms are awesome, they let you test logic without testing the whole view code.